### PR TITLE
Update Metric Limiter with latest release

### PR DESCRIPTION
**Pull request by harrryr**
*Issue #, if available:*
New Java Artifacts will be released. The new release includes significant changes in the metric schema of the telemetry. We need to update the validation templates to reflect the new expected changes. 

May 30, 2024 Update:
Java Artifacts have been released.
- Updated file name from appsignals -> application-signals
- Removed code forcing old version of artifact

Test run: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/9322261852
Second run: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/9322363457
Third run: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/9324310243

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/application-signals-java-e2e-metric-limiter-test.yml [hunk 1]
```diff
@@ -176,9 +170,8 @@ jobs:
                 --region ${{ inputs.aws-region }} \
                 --configuration-values '{"agent":{"config":{"logs":{"metrics_collected":{"app_signals":{"limiter":{"drop_threshold":2}}}}}}}'
                     
-          
-              execute_and_retry 2 "kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}" "" 60
-              execute_and_retry 2 "kubectl wait --for=condition=Ready --request-timeout '5m' pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}" "" 10
+              execute_and_retry 2 "kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}"
+              execute_and_retry 2 "kubectl wait --for=condition=Ready --request-timeout '5m' pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}"
```
Comment (majanjua-amzn) [inline .github/workflows/application-signals-java-e2e-metric-limiter-test.yml:174]: Unnecessary change

File: .github/workflows/application-signals-java-e2e-metric-limiter-test.yml [hunk 2]
```diff
@@ -279,10 +272,10 @@ jobs:
...
         continue-on-error: true
         run: |
-          curl -S -s "http://${{ env.APP_ENDPOINT }}/outgoing-http-call"; echo
-          curl -S -s "http://${{ env.APP_ENDPOINT }}/aws-sdk-call"; echo
-          curl -S -s "http://${{ env.APP_ENDPOINT }}/remote-service?ip=${{ env.REMOTE_SERVICE_POD_IP }}"; echo
-          curl -S -s "http://${{ env.APP_ENDPOINT }}/client-call"; echo
+          curl -S -s "http://${{ env.APP_ENDPOINT }}/outgoing-http-call"
+          curl -S -s "http://${{ env.APP_ENDPOINT }}/aws-sdk-call?ip=${{ env.REMOTE_SERVICE_POD_IP }}&testingId=${{ env.TESTING_ID }}"
+          curl -S -s "http://${{ env.APP_ENDPOINT }}/remote-service?ip=${{ env.REMOTE_SERVICE_POD_IP }}&testingId=${{ env.TESTING_ID }}"
+          curl -S -s "http://${{ env.APP_ENDPOINT }}/client-call"
```
Comment (majanjua-amzn) [inline .github/workflows/application-signals-java-e2e-metric-limiter-test.yml:278]: Put the echos back

Comment (harrryr) [reply .github/workflows/application-signals-java-e2e-metric-limiter-test.yml]: Done

File: .github/workflows/application-signals-java-e2e-metric-limiter-test.yml (hunk 1 above)
Comment (harrryr) [reply .github/workflows/application-signals-java-e2e-metric-limiter-test.yml]: Done

**Review by majanjua-amzn — APPROVED** [review]
